src/main/scala/com/timeout/KinesisGraphStage.scala

```diff
@@ -70,6 +74,26 @@ class KinesisGraphStage[A : ToPutRecordsRequest](putRecords: PutRecords, streamN
       }
     }
 
+    /**
+      * Split
+      */
+    private def splitBySize(entries: List[PutRecordsRequestEntry]): List[List[PutRecordsRequestEntry]] = {
+      val zero = 0 -> List.empty[PutRecordsRequestEntry] -> List.empty[List[PutRecordsRequestEntry]]
+      entries.foldLeft(zero) { case (((total, newList), lists), entry) =>
+        val size = entry.getData.limit() // an approximation, as in the request it will have different size
+        if (total + size > maxRequestSize)
```

Comment:
Ok, so here is a caveat. Size of string != size of record. We may be adding some metadata later to the message, so it can be bigger in the end? Then we shouldn't pack up to `maxRequestSize` of bytes, and instead like 90% of `maxRequestSize`? Should I just hardcode it or make it configurable?